python/paddle/fluid/tests/unittests/test_multi_tensor_adam.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import paddle
+import paddle.nn as nn
+import numpy as np
+
+
+class MLPLayer(nn.Layer):
+
+    def __init__(self, input_size, hidden_size, output_size, n):
+        super(MLPLayer, self).__init__()
+        self.linear_first = nn.Linear(input_size, hidden_size)
+        self.linear_mid = nn.Sequential(
+            ('l0', nn.Linear(hidden_size, hidden_size)))
+        for i in range(n - 1):
+            self.linear_mid.add_sublayer('l' + str(i + 1),
+                                         nn.Linear(hidden_size, hidden_size))
+        self.linear_last = nn.Linear(hidden_size, output_size)
+
+    def forward(self, x):
+        x = self.linear_first(x)
+        x = self.linear_mid(x)
+        x = self.linear_last(x)
+        return x.mean()
+
+
+class TestMultiTensorAdam(unittest.TestCase):
+
+    def setUp(self):
+        paddle.disable_static()
+        self.input_size = 800
+        self.hidden_size = 500
+        self.output_size = 700
+        self.n = 10
+
+    def get_multi_tensor_adam(self, mode):
+
+        paddle.seed(10)
+
+        model = MLPLayer(self.input_size, self.hidden_size, self.output_size,
+                         self.n)
+
+        inp = paddle.uniform([10, self.input_size], dtype="float32", seed=10)
+
+        out = model(inp)
+
+        beta1 = paddle.to_tensor([0.9], dtype="float32")
+        beta2 = paddle.to_tensor([0.99], dtype="float32")
+
+        opt = paddle.optimizer.MultiTensorAdam(learning_rate=0.1,
+                                               parameters=model.parameters(),
+                                               weight_decay=0.01,
+                                               beta1=beta1,
+                                               beta2=beta2,
+                                               mode=mode)
+        out.backward()
+        opt.step()
+        opt.clear_grad()
+
+        return model.parameters()
+
+    def get_adam_or_adamw(self, mode):
```

Comment:
Too many duplicate codes between `get_multi_tensor_adam` and `get_adam_or_adamw`. Try to unify the codes.